version: "3.3"
services:
  bot:
    build:
      context: .
    working_dir: /home/python_user/app/bot
    env_file: ../../.env
    restart: always
    command: bash -c "./run.sh"
    tty: true
    volumes:
      - ./:/home/python_user/app/bot
      - ../modules/:/home/python_user/app/modules
      - venvs:/home/python_user/.local/share/pdm/venvs
      - ~/logs:/home/python_user/logs
      - ../../pyproject.toml:/home/python_user/app/pyproject.toml
      - ../../pdm.lock:/home/python_user/app/pdm.lock
      - ../../alembic.ini:/home/python_user/app/alembic.ini
    networks:
      - net

  redis:
    image: "redis:latest"
    restart: always
    ports:
      - "6379:6379"
    volumes:
      - redis_data:/data
    command: ["redis-server", "--appendonly", "yes"]
    networks:
      - net

volumes:
  venvs:
  redis_data:
    driver_opts:
      type: none
      o: bind
      device: /home/bot/redis_data

networks:
  net:
    driver: bridge
